proto/cosmos/tx/v1beta1/service.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+package cosmos.tx.v1beta1;
+
+import "google/api/annotations.proto";
+import "cosmos/base/abci/v1beta1/abci.proto";
+import "cosmos/tx/v1beta1/tx.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/types/tx";
+
+// Service defines a gRPC service for simulating transactions.
+service Service {
+    // Simulate simulates executing a transaction for estimating gas usage.
+    rpc Simulate(SimulateRequest) returns (SimulateResponse) {
+        option (google.api.http).post = "/cosmos/tx/v1beta1/simulate";
+    }
+    // Simulate simulates executing a transaction for estimating gas usage.
+    rpc GetTx(GetTxRequest) returns (GetTxResponse) {
+        option (google.api.http).get = "/cosmos/tx/v1beta1/tx/{hash}";
+    }
```

Comment:
This method is a simple proxy to TM's `/tx` rpc endpoint. I'm wondering if long-term, it's a good idea to leave it in this Tx service.

Once we remove legacy REST endpoints, people might ask for a grpc-gateway versions of `/node_info`, `/blocks/{height}` and other [TM rpc endpoints](https://github.com/cosmos/cosmos-sdk/blob/7792ccf3420aa41a1e54dc9ad679ea5bd7b23625/client/rpc/routes.go#L11-L16). Where should we put those methods?

We could alternatively expose a big `Tendermint` gRPC service, which acts as a proxy service to TM methods defined [here](https://docs.tendermint.com/master/rpc/). In particular, this `GetTx` method would go there. This effort should be coordinated with TM's own plan to expose its RPC methods via Protobuf.

What do you think?

Reply:
The thinking here is to provide gRPC endpoints for things that use our `Tx` type. `/node_info`, etc. don't use our `Tx` type so would naturally be separate services. Likely we should coordinate that with the Tendermint team. /cc @marbar3778